sendgrid/helpers/mail/mail.py

```diff
@@ -308,14 +308,40 @@ def attachments(self):
         """
         return self._attachments
 
-    def add_attachment(self, attachment):
+    def add_attachment(self, mail_attachment):
         """Add an Attachment to this Mail.
 
         :type attachment: Attachment
         """
         if self._attachments is None:
             self._attachments = []
-        self._attachments.append(attachment)
+        if isinstance(mail_attachment, attachment.S3Attachment):
+            self.download_s3_attachment(mail_attachment)
+        self._attachments.append(mail_attachment)
+
+    def download_s3_attachment(self, s3_attachment):
+        """
+        Requires boto3, botocore to be installed
+        :param s3_attachment: S3Attachment
+        :return:
+        """
+        import boto3
+        import botocore
+        import base64
+        s3 = boto3.resource('s3') if s3_attachment.session is None else s3_attachment.session.resource('s3')
```

Comment:
This line is 105 characters long and since one of the other issues is making this repo PEP8 compliant, maybe you could just use a regular `if/else` instead of this long ternary?

```python
    if s3_attachment.session is None:
        s3 = boto3.resource('s3')
    else:
        s3 = s3_attachment.session.resource('s3')
```

Reply:
Acknowledged, made the necessary changes.